zetaclient/chains/evm/signer/v2_sign.go

```diff
@@ -29,19 +29,18 @@ func (signer *Signer) signGatewayExecute(
 
 	var data []byte
 
+	// if sender is provided in messageContext call is authenticated and target is Callable.onCall
```

Comment:
I think the function in `v2_sign.go` should be the minimal bricks that directly pack and sign the requested function without logic how the signed data is structure.

I think this logic should be in the function where we create `txData *OutboundData`